### PR TITLE
Parse config type aliases

### PR DESCRIPTION
**Pull request by alsamylkin**
In convertValue we have a type switch that is using output direct type, thus failing to set value if a target type is e.g. an alias. Shawn proposed a fix https://github.com/uber-go/fx/pull/424, but we should keep numerical parsing in one place and remove it from generic convertValue function.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -426,3 +426,128 @@ func TestNopProvider_Get(t *testing.T) {
...
+
+	require.NoError(t, v.Populate(cfg))
+}
+
+func TestPointerChildTypedField(t *testing.T) {
+	t.Parallel()
+
+	type Port int
+	type childPort struct {
+		Port *Port
```
Comment (anuptalwalkar) [inline config/config_test.go]: test with non pointer type?

File: config/value.go [hunk 2]
```diff
@@ -282,10 +281,6 @@ func convertValue(value interface{}, targetType reflect.Type) (interface{}, erro
 	case string:
 		target := reflect.New(targetType).Interface()
 		switch t := target.(type) {
-		case *int:
-			return strconv.Atoi(v)
-		case *bool:
-			return strconv.ParseBool(v)
 		case *time.Duration:
```
Comment (anuptalwalkar) [inline config/value.go]: what if time.Duration has an alias? Should we add test for it?

File: config/config_test.go (hunk 1 above)
Comment (alsamylkin) [inline config/config_test.go]: We already have plenty of them, e.g.: https://github.com/uber-go/fx/blob/dev/config/decoder_test.go#L233

This is a regression test from Shawn's PR.

File: config/value.go (hunk 2 above)
Comment (alsamylkin) [inline config/value.go]: Well, time.Duration is an alias https://golang.org/pkg/time/#Duration :)

**Pull request merged**